lib/board.rb

```diff
@@ -2,32 +2,44 @@ class Board
   attr_reader :grid
 
   def initialize
-    @grid = top_row
+    @grid = numbered_row
     build_grid
-    add_numbers
   end
 
   def to_s
-    grid.to_s
+    grid
   end
 
   private
 
-  def row
-    "|  |  |  |\n|__|__|__|\n".freeze
+  def numbered_row
+    "   1  2  3\n".freeze
   end
 
-  def top_row
-    " __ __ __\n".freeze
+  def build_grid
+    add_top_grid_row
+    add_grid_rows
   end
 
-  def build_grid
-    3.times do
-      @grid += row
+  def add_top_grid_row
+    @grid += top_grid_row
+  end
+
+  def top_grid_row
+    "   __ __ __\n".freeze
+  end
+
+  def add_grid_rows
+    3.times do |i|
+      @grid = grid + column_labels[i] + row
```

Comment:
I think there's a mixture of concerns here and somewhat of a focus on the underlying data instead of message passing.

Concerns - This method knows about the overall grid, how to generate a column label, that a row follows it, and that it all happens three times.

Data - The indexing into `column_labels` stands out as a focus on the array, while the other parts are pretty well defined names about the pieces required to build up the grid.

Reply:
This is a good point. I agree that `add_grid_rows` is doing a lot and has access to grid and labels and row. I tried to refactor to address that. 

I also tried to address the message vs. data bit you're pointing out. As I understand it, what we mean by messages is like how these methods are named for what they are and directly update `grid`. Now we have methods for constructing each row and then another to add them together. I'm not sure if that addresses what you're pointing out, open to additional feedback!